packages/provider/src/tests/dataUtils/funds.ts

```diff
@@ -66,39 +68,69 @@ export async function sendFunds(
+            const unsub = await api.tx.balances
+                .transferAllowDeath(address, amount)
+                .signAndSend(pair, { nonce }, (result: ISubmittableResult) => {
+                    if (result.status.isInBlock || result.status.isFinalized) {
+                        result.events
+                            .filter(({ event: { section } }: any): boolean => section === 'system')
+                            .forEach((event): void => {
+                                const {
+                                    event: { method },
+                                } = event
 
-                            if (method === 'ExtrinsicFailed') {
-                                unsub()
-                                reject(dispatchErrorHandler(api.registry, event))
-                            }
-                        })
-                    unsub()
-                    resolve(result)
-                } else if (result.isError) {
-                    unsub()
-                    reject(result)
-                }
-            })
-    })
-    await result
-        .then((result: ISubmittableResult) => {
-            env.logger.debug(who, 'sent amount', unitAmount, 'UNIT at tx hash ', result.status.asInBlock.toHex())
+                                if (method === 'ExtrinsicFailed') {
+                                    unsub()
+                                    reject(dispatchErrorHandler(api.registry, event))
+                                }
+                            })
+                        unsub()
+                        resolve(result)
+                    } else if (result.isError) {
+                        unsub()
+                        reject(result)
+                    }
+                })
         })
-        .catch((error) => {
-            throw new ProsopoEnvError('DEVELOPER.FUNDING_FAILED', { context: { error } })
+    } else {
+        result = new Promise((resolve, reject) => {
+            try {
+                const extrinsic = api.tx.balances.transferAllowDeath(address, amount)
+                txQueue.add(
+                    extrinsic,
+                    (txResult: ISubmittableResult) => {
+                        env.logger.info('In sendFunds call back')
+                        resolve(txResult)
+                    },
+                    pair
+                )
+            } catch (e) {
+                reject(e)
+            }
         })
+    }
+    const submittableResult = await result
+
+    if (submittableResult.isError) {
+        throw new ProsopoEnvError('DEVELOPER.FUNDING_FAILED', {
+            context: {
+                error: submittableResult.dispatchError
+                    ? submittableResult.dispatchError.toHuman()
+                    : submittableResult.toHuman(),
+            },
+        })
+    }
+    if (submittableResult.status.isFuture) {
+        const block = await api.rpc.chain.getBlock()
+        console.log('Current block', JSON.stringify(block))
```

Comment:
how come console.log + env.logger.debug? are they logging in different environments?

Reply:
Shouldn't be there. Removed.